src/io/hdf/hdf_path.cpp

```diff
@@ -23,8 +23,13 @@ hdf_path& hdf_path::append(const hdf_path& p)
 {
   if (p.has_root_directory())
     path_.clear();
-  else if (path_.back() != '/')
```

Comment:
I can understand that accessing back() without protecting the empty string case can cause invalid access. I don't understand why asan still fails after adding a check of `empty`. The even more strange thing is that I cannot reproduce the asan failure locally. I was using 11.2 but Paul has 11.3. I'm building 11.4 now.

Reply:
FYI, these builds were with RHEL 9.2 entirely using RHEL provided packages for the compiler, libraries etc.